sprint_8-solution-time-and-duration

Verba163 opened the pull request:
1. Добавить классы startTime, duration, endTime, для классов Task, Subtask, Epic
2. Создать сортировщик для списка задач с приоритетом в дату старт задачи
3. Переписать методы, которые возможно через Stream API
4. Переработать класс Сохранения/Загрузки под новый формат задач(время)
5. Переработать интерфейс TaskManagers под новые поля задач(время)
6. Переработать Main для тестов вывода отсортированного списка задач
7. Разработать проверку на пересечение времени задач, реализовать для нее Junit проверку
8. Переработать Junit тесты под новые типы задач

Verba163 committed:
1. Добавить классы startTime, duration, endTime, для классов Task, Subtask, Epic
2. Создать сортировщик для списка задач с приоритетом в дату старт задачи
3. Переписать методы, которые возможно через Stream API
4. Переработать класс Сохранения/Загрузки под новый формат задач(время)
5. Переработать интерфейс TaskManagers под новые поля задач(время)
6. Переработать Main для тестов вывода отсортированного списка задач
7. Разработать проверку на пересечение времени задач, реализовать для нее Junit проверку
8. Переработать Junit тесты под новые типы задач

Verba163 committed:
Исправить чекстайл

Verba163 committed:
Исправить чекстайл

Verba163 committed:
Исправить чекстайл

felixoff commented on src/ru/yandex/practicum/managers/InMemoryTaskManager.java:
```diff
@@ -6,60 +6,88 @@
+
+        Subtask subtask = new Subtask(title, description, taskId, duration, startTime);
+        if (isSubtaskOverlapping(subtask)) {
+            throw new IllegalArgumentException("Подзадача пересекается с существующей задачей или подзадачей.");
```
Какая с какой? Указывает конкретную инфу о задаче

felixoff commented on src/ru/yandex/practicum/models/Task.java:
```diff
@@ -2,23 +2,31 @@
 
-public class Task {
+public class Task implements Comparable<Task> {
+    //static DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy HH:mm");
```
Не оставляй закоментированные ку, ки в чистовом коде

Verba163 committed:
1. Исправить выброс исключения -> исключение выводит конфликтующие таски
2. Переписать OverLapTests
3. Добавить метод получения epicId
4. Удалить неиспользуемые закомментированные строки
5. Переписать методы Overlap

Verba163 committed:
Исправить чекстайл
